src/gazebo_mavlink_interface.cpp

```diff
@@ -538,192 +600,238 @@ void GazeboMavlinkInterface::OnUpdate(const common::UpdateInfo&  /*_info*/) {
-  // frames
-  // g - gazebo (ENU), east, north, up
-  // r - rotors imu frame (FLU), forward, left, up
-  // b - px4 (FRD) forward, right down
-  // n - px4 (NED) north, east, down
-  math::Quaternion q_gr = math::Quaternion(
+  common::Time current_time = world_->GetSimTime();  
+  double dt = (current_time - last_imu_time_).Double();
+  if (dt >= imu_rate_)
+  {
+    // frames
+    // g - gazebo (ENU), east, north, up
+    // r - rotors imu frame (FLU), forward, left, up
+    // b - px4 (FRD) forward, right down
+    // n - px4 (NED) north, east, down
+    math::Quaternion q_gr = math::Quaternion(
       imu_message->orientation().w(),
       imu_message->orientation().x(),
       imu_message->orientation().y(),
       imu_message->orientation().z());
 
-  // q_br
-  /*
-     tf.euler2quat(*tf.mat2euler([
-   #        F  L  U
-          [1, 0, 0],  # F
-          [0, -1, 0], # R
-          [0, 0, -1]  # D
-      ]
-     )).round(5)
-   */
-  math::Quaternion q_br(0, 1, 0, 0);
-
-
-  // q_ng
-  /*
-     tf.euler2quat(*tf.mat2euler([
-   #        N  E  D
-          [0, 1, 0],  # E
-          [1, 0, 0],  # N
-          [0, 0, -1]  # U
-      ]
-     )).round(5)
-   */
-  math::Quaternion q_ng(0, 0.70711, 0.70711, 0);
-
-  math::Quaternion q_gb = q_gr * q_br.GetInverse();
-  math::Quaternion q_nb = q_ng * q_gb;
-
-  math::Vector3 pos_g = model_->GetWorldPose().pos;
-  math::Vector3 pos_n = q_ng.RotateVector(pos_g);
-
-  //gzerr << "got imu: " << C_W_I << "\n";
-  //gzerr << "got pose: " << T_W_I.rot << "\n";
-  float declination = get_mag_declination(groundtruth_lat_rad, groundtruth_lon_rad);
-
-  math::Quaternion q_dn(0.0, 0.0, declination);
-  math::Vector3 mag_n = q_dn.RotateVector(mag_d_);
-
-  math::Vector3 vel_b = q_br.RotateVector(model_->GetRelativeLinearVel());
-  math::Vector3 vel_n = q_ng.RotateVector(model_->GetWorldLinearVel());
-  math::Vector3 omega_nb_b = q_br.RotateVector(model_->GetRelativeAngularVel());
-
-  math::Vector3 mag_noise_b(
+    // q_br
+    /*
+    tf.euler2quat(*tf.mat2euler([
+    #        F  L  U
+            [1, 0, 0],  # F
+            [0, -1, 0], # R
+            [0, 0, -1]  # D
+        ]
+    )).round(5)
+    */
+    math::Quaternion q_br(0, 1, 0, 0);
+
+
+    // q_ng
+    /*
+    tf.euler2quat(*tf.mat2euler([
+    #        N  E  D
+            [0, 1, 0],  # E
+            [1, 0, 0],  # N
+            [0, 0, -1]  # U
+        ]
+    )).round(5)
+    */
+    math::Quaternion q_ng(0, 0.70711, 0.70711, 0);
+
+    math::Quaternion q_gb = q_gr*q_br.GetInverse();
+    math::Quaternion q_nb = q_ng*q_gb;
+
+    math::Vector3 pos_g = model_->GetWorldPose().pos;
+    math::Vector3 pos_n = q_ng.RotateVector(pos_g);
+
+    //gzerr << "got imu: " << C_W_I << "\n";
+    //gzerr << "got pose: " << T_W_I.rot << "\n";
+    float declination = get_mag_declination(groundtruth_lat_rad, groundtruth_lon_rad);
+
+    math::Quaternion q_dn(0.0, 0.0, declination);
+    math::Vector3 mag_n = q_dn.RotateVectorReverse(mag_d_);
```

Comment:
This is reversing the change of https://github.com/PX4/sitl_gazebo/commit/1dc976f89d3bee46c3c29e8f4608ca7af9a49412